### ISSUE TITLE
1.1b1 fails on test_hist_autobin

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#4329]
When buildng the 1.1b1 release, the `test_hist_autobin` function fails on Python 2.7:
```
=================================== FAILURES ===================================
______________________________ test_hist_autobin _______________________________

rseed = 0

    @pytest.mark.skipif('not HAS_PLT')
    def test_hist_autobin(rseed=0):
        rng = np.random.RandomState(rseed)
        x = rng.randn(100)
    
        for bintype in [10, np.arange(-3, 3, 10),
                        'knuth', 'scott', 'freedman', 'blocks']:
            for range in [None, (-3, 3)]:
                n1, bins1 = histogram(x, bintype, range=range)
>               n2, bins2, patches = hist(x, bintype, range=range)

astropy/visualization/tests/test_histogram.py:57: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/visualization/hist.py:63: in hist
    return ax.hist(x, bins, **kwargs)
/usr/lib/python2.7/dist-packages/matplotlib/__init__.py:1808: in inner
    return func(ax, *args, **kwargs)
/usr/lib/python2.7/dist-packages/matplotlib/axes/_axes.py:6082: in hist
    color=c, **{bottom_kwarg: bottom})
/usr/lib/python2.7/dist-packages/matplotlib/__init__.py:1808: in inner
    return func(ax, *args, **kwargs)
self = <matplotlib.axes._subplots.AxesSubplot object at 0x7fe18db4ecd0>
left = array([], dtype=float64), height = array([], dtype=float64)
width = array([], dtype=float64), bottom = array([], dtype=float64), kwargs = {}
color = [array([ 0.  ,  0.75,  0.75,  1.  ])], edgecolor = []
linewidth = [None], tick_label = [None], label_ticks_flag = False, xerr = None
yerr = None

    @unpack_labeled_data(replace_names=["left", "height", "width", "bottom",
                                        "color", "edgecolor", "linewidth",
                                        "tick_label", "xerr", "yerr",
                                        "ecolor"],
                         label_namer=None)
    @docstring.dedent_interpd
    def bar(self, left, height, width=0.8, bottom=None, **kwargs):
        """[...]"""
    [...]
        if len(tick_label) != nbars:
            raise ValueError("incompatible sizes: argument 'tick_label' "
>                            "must be length %d or string" % nbars)
E           ValueError: incompatible sizes: argument 'tick_label' must be length 0 or string

/usr/lib/python2.7/dist-packages/matplotlib/axes/_axes.py:2100: ValueError
```

**Comment by astrofrog** [comment]
@olebole - what version of Matplotlib are you testing with?

**Comment by olebole** [comment]
Matplotlib `1.5.0~rc2-1`

**Comment by embray** [comment]
I don't think we've tested that.  Seems almost more like a matplotlib issue than anything.

**Comment by astrofrog** [comment]
We do test the latest stable Matplotlib on Travis in the build with optional dependencies but this wasn't failing. I also can't reproduce locally. @olebole - can you try the final 1.5 release to see if that changes anything?

**Comment by olebole** [comment]
I rely here on what is packaged for Debian, and the matplotlib team still didn't upload this yet :-(

**Comment by eteq** [comment]
I just tested this right now (with Ubuntu 15.10) on both matplotlib 1.5.0rc2 and matplotlib 1.5.0 (final), and I *do* see the same error with 1.5.0rc2 and *not* with 1.5.0. So I conclude that this wasa matplotlib bug that was fixed in 1.5.0 final.  So I'll close this for now.

But @olebole, if it turns out to be time to package the astropy v1.1 final release and there's still no 1.5.0 final for matplotlib, is it permissible to just skip that particular test in the version that gets packaged with debian.  Presumably it would be a once-off, as by the time another astropy release came there should be an mpl 1.5.0 package.
